### PR TITLE
HWKMETRICS-84 Update ptrans to use tenant as a request header

### PR DESCRIPTION
**Pull request by tsegismont**
Ready for review.

### CODE REVIEW

File: clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java [hunk 1]
```diff
@@ -94,7 +96,7 @@
     @Before
     public void setUp() throws Exception {
         tenant = getRandomTenantId();
-        findGuageMetricsUrl = "http://" + BASE_URI + "/metrics?type=gauge&tenantId=" + tenant;
+        findGaugeMetricsUrl = "http://" + BASE_URI + "/metrics?type=gauge";
```
Comment (pilhuhn) [inline clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java]: No tenant needed here?

File: clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java [hunk 2]
```diff
@@ -242,12 +245,13 @@ private Point collectdLogToPoint(String line) {
     private List<Point> getServerData() throws Exception {
         ObjectMapper objectMapper = new ObjectMapper();
 
-        HttpURLConnection urlConnection = (HttpURLConnection) new URL(findGuageMetricsUrl).openConnection();
+        HttpURLConnection urlConnection = (HttpURLConnection) new URL(findGaugeMetricsUrl).openConnection();
+        urlConnection.setRequestProperty("tenantId", tenant);
```
Comment (pilhuhn) [inline clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java]: Also use `TENANT.getExternalForm()`?

File: clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java [hunk 3]
```diff
@@ -264,6 +268,7 @@ private Point collectdLogToPoint(String line) {
             String type = split[split.length - 1];
 
             urlConnection = (HttpURLConnection) new URL(findGaugeDataUrl(metricName)).openConnection();
+            urlConnection.setRequestProperty("tenantId", tenant);
```
Comment (pilhuhn) [inline clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java]: I think it is a bit confusing to me that you sometimes use `TENANT` and sometimes `tenantId`. While I see that the former is used for the config properties, it still confuses me. Perhaps not using a static import would help me here.

Btw: why did you introduce `getExternalForm()` and not override `toString()`?

**Comment by pilhuhn** [comment]
Lgtm - my comment is more of a cosmetic nature

File: clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java (hunk 3 above)
Comment (tsegismont) [inline clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java]: Le 12/05/2015 13:06, Heiko W. Rupp a écrit :
> I think it is a bit confusing to me that you sometimes use |TENANT| and
> sometimes |tenantId|. While I see that the former is used for the config
> properties, it still confuses me. Perhaps not using a static import
> would help me here.

I'll create another constant for the header name and won't use the 
static import for config param name.

>
> Btw: why did you introduce |getExternalForm()| and not override
> |toString()|?
>

Maybe I should have. This enum purpose is precisely to map config param 
names. So #toString may be more natural. Is that your thinking?

Comment (pilhuhn) [inline clients/ptranslator/src/test/java/org/hawkular/metrics/clients/ptrans/fullstack/CollectdITest.java]: On 12 May 2015, at 14:06, Thomas Segismont wrote:
> names. So #toString may be more natural. Is that your thinking?

Yes. And can even be omitted, as the implicit call to `toString()` would
then do the right thing already.

**Pull request merged**